libfaketime/__init__.py

```diff
@@ -103,17 +104,23 @@ def __init__(self, datetime_spec, only_main_thread=True):
             _datetime = dateutil.parser.parse(datetime_spec)
 
         self.time_to_freeze = _datetime  # freezegun compatibility
-        self.libfaketime_spec = _datetime.strftime('%Y-%m-%d %T %f')
 
     def _should_fake(self):
         return self.only_main_thread and threading.current_thread().name == 'MainThread'
 
-    def __enter__(self):
+    def _start_fake(self):
+        self.libfaketime_spec = self.time_to_freeze.strftime('%Y-%m-%d %T %f')
```

Comment:
Any reason to keep this an instance field? Looks like it might be able to just be a local variable now.